src/interchange/src/protobuf.rs

```diff
@@ -55,9 +61,14 @@ fn validate_proto_field(field: &FieldDescriptor, descriptors: &Descriptors) -> R
                 FieldType::String => ScalarType::String,
                 FieldType::Bytes => ScalarType::Bytes,
                 FieldType::Message(m) => {
+                    if seen_messages.contains(m.name()) {
+                        bail!("Recursive messages are not supported: {}", m.name());
```

Comment:
In avro.rs we use:

```
                        bail!(
                            "Recursively defined type in schema: {}",
                            v.get_human_name(schema.root)
                        );
```

We should probably make the messages match.

Reply:
Ack, changed both to "Recursive types are not supported: {}". Error messages that are statements of fact ("Recursive type in schema: {}") are a bit of a pet peeve of mine because they only _imply_ that the error is that recursive types are unsupported; they don't tell you outright. 

I think the best error message here would be one that uses the new detail/hint setup, because the current phrasing is still awkward. Something like:

```
ERROR: recursive type detected in schema
DETAIL: The following cycle was detected:
  Message1
  Message2
  Message3
  Message1
```

But, not gonna shave that yak presently.